packages/tabs-react/src/Tabs.tsx

```diff
@@ -0,0 +1,187 @@
+import React, { ReactNode, useLayoutEffect, useRef, useState, FC, useCallback, useEffect } from "react";
+
+/**
+ * Konteiner for innhold i et tabbet brukergrensesnitt. Ikke funksjonell utenfor et Tabs element.
+ *
+ * Docs: https://jokul.fremtind.no/komponenter/tabs
+ */
+export const TabPanel: FC<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>> = ({
+    children,
+    ...props
+}) => {
+    return (
+        <div role="tabpanel" {...props}>
+            {children}
+        </div>
+    );
+};
+
+interface TabProps {
+    children: string;
+}
+
+/**
+ * En Tab til bruk som element i TabList. Ikke funksjonell utenfor et TabList element.
+ *
+ * Docs: https://jokul.fremtind.no/komponenter/tabs
+ */
+// eslint-disable-next-line react/display-name
+export const Tab = React.forwardRef<HTMLButtonElement, TabProps>((props, ref) => (
+    <button className="jkl-tab" role="tab" type="button" ref={ref} {...props} />
+));
+
+interface TabListProps {
```

Comment:
Jeg liker å extende React-typene for HTML-elementet vi bruker som rotelement for å få typet ferdig `children`, `className` og alt av native HTML-props.

```suggestion
interface TabListProps extends HTMLAttributes<HTMLDivElement> {}
```

Reply:
Dette er kanskje ikke det riktige stedet for å diskutere det, men jeg er litt usikker på om det bør være en universell tilnærming? Det åpner jo for å gjøre mye som kanskje ikke er tiltenkt bruk av komponenten, og det gjør IntelliSense ganske slitsomt å bruke siden du plutselig for 250 forslag til props 😅

Er veldig for å gjøre det på komponenter som er rene erstatninger for et eksisterende HTML-element, som f.eks. tabellkomponentene eller knapper. Men jeg klarer ikke helt å se grunnen til at `TabList` skal kunne brukes som et generelt `div`-element? Komponenten har jo en veldig spesifikk funksjon her.

I dette tilfellet kunne man kanskje heller extendet `AriaProperties` og typet `TabList` som en `FunctionalComponent` hvis man vil slippe å spesifisere `children` og `aria-label`.

Reply:
Du har et poeng @piofinn. Det er ikke alltid man vil akseptere hva som helst som children heller. I dette tilfellet er det heller ingen mening i å gi elementet noen annen aria property enn aria-label som er spesifisert.